apps/accelerate/chatllama/chatllama/llama_model.py

```diff
@@ -177,17 +597,38 @@ def load_model(
     tokenizer_path: str,
     local_rank: int,
     world_size: int,
+    froze_embeddings: bool,
+    use_fairscale: bool,
     max_batch_size: int = 32,
 ) -> Tuple[Transformer, HFLikeTokenizer]:
-    checkpoint, params = load_checkpoints(ckpt_dir, local_rank, world_size)
-    model_args: ModelArgs = ModelArgs(
-        max_seq_len=1024, max_batch_size=max_batch_size, **params
-    )
-    tokenizer = Tokenizer(model_path=tokenizer_path)
-    model_args.vocab_size = tokenizer.n_words
-    torch.set_default_tensor_type(torch.cuda.HalfTensor)
-    model = Transformer(model_args)
-    torch.set_default_tensor_type(torch.FloatTensor)
-    model.load_state_dict(checkpoint, strict=False)
-    tokenizer = HFLikeTokenizer(tokenizer)
+
+    # test the model with hf tokenizer
+    test = True
+    if not test:
+        checkpoint, params = load_checkpoints(ckpt_dir, local_rank, world_size)
+        model_args: ModelArgs = ModelArgs(
+            max_seq_len=1024, max_batch_size=max_batch_size, **params
+        )
+        checkpoint, params = load_checkpoints(ckpt_dir, local_rank, world_size)
+        model_args: ModelArgs = ModelArgs(
+            max_seq_len=1024, max_batch_size=max_batch_size, **params
+        )
+        model_args.forze_embeddings = froze_embeddings
+        model_args.use_fairscale = use_fairscale
+        tokenizer = Tokenizer(model_path=tokenizer_path)
+        model_args.vocab_size = tokenizer.n_words
+        torch.set_default_tensor_type(torch.cuda.HalfTensor)
+        model = Transformer(model_args)
+        torch.set_default_tensor_type(torch.FloatTensor)
+        model.load_state_dict(checkpoint, strict=False)
+        tokenizer = HFLikeTokenizer(tokenizer)
```

Comment:
I think that this section should be moved somewhere else

Reply:
what do you mean?